### PR TITLE
Added playbook to install meta operator

### PR DESCRIPTION
**Pull request by raukadah**
https://github.com/openstack-k8s-operators/ci-framework/pull/201 changes the parent from base-crc to base-simple-crc.

base-simple-crc job will provide an crc based openshift cluster. It lacks the meta operator installation bits which breaks the existing job.

https://github.com/openstack-k8s-operators/ci-framework/pull/245 refines the cifmw-base-crc-openstack job and moves the ci-framework pre playbooks in pre-run and collect-logs in post-run playbook. It will help to drop the existing collect-logs.yaml playbook.

In order to fix the job, It uses install_yamls_make role to install meta operator.

Depends-On: https://github.com/openstack-k8s-operators/ci-framework/pull/286

### CODE REVIEW

**Comment by SeanMooney** [comment]
we could probably proceed with this once the ci issues are addressed

I actually have already started working on the same change but in a different way https://github.com/openstack-k8s-operators/nova-operator/compare/main...SeanMooney:nova-operator:zuul-jobs-part-2

effectively i was trying to figure out how to install the meta operator without install_yamls but i still need to figure out some of the network isolation bits.

File: ci/nova-operator-base/playbooks/install_meta_operators.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  ansible.builtin.import_playbook: "{{ ci_framework_basedir }}/ci_framework/playbooks/01-bootstrap.yml"
+
+- name: Install meta operator
+  hosts: localhost
+  tasks:
+    - name: Run make download_tools
+      ansible.builtin.include_role:
+        name: 'install_yamls_makes'
+        tasks_from: 'make_download_tools'
+
```
Comment (SeanMooney) [inline ci/nova-operator-base/playbooks/install_meta_operators.yaml:12]: we need to add the addtional interface to CRC before we do make OpenStack

that is why its failing waiting for enp6s0

File: ci/nova-operator-base/playbooks/pre.yaml [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
+    ci_framework_basedir: "{{ ansible_user_dir }}/src/github.com/openstack-k8s-operators/ci-framework"
   tasks:
     - name: Add crc creds
       ansible.builtin.include_role:
         name: rhol_crc
         tasks_from: add_crc_creds
 
+    - name: Install meta operator
+      ansible.builtin.shell: |
+        export ANSIBLE_CONFIG={{ ansible_user_dir }}/src/github.com/openstack-k8s-operators/ci-framework/ansible.cfg
```
Comment (SeanMooney) [inline ci/nova-operator-base/playbooks/pre.yaml:16]: export ANSIBLE_CONFIG={{ ansible_user_dir }}/src/github.com/openstack-k8s-operators/ci-framework/ansible.cfg

can be

export ANSIBLE_CONFIG={{ ci_framework_basedir }}/ansible.cfg

File: ci/nova-operator-base/playbooks/pre.yaml [hunk 3]
```diff
@@ -1,19 +1,23 @@
...
+        ansible-playbook install_meta_operators.yaml -e ci_framework_basedir={{ ci_framework_basedir }}
```
Comment (SeanMooney) [inline ci/nova-operator-base/playbooks/pre.yaml:17]: in generall i dont really like the pattern of invoking a playbook via a shell like this.

this feels like an anisble anti pattern to me 

i would prefer to do an import

- name: Install meta operator
  ansible.builtin.import_playbook: { nova_operator_basedir }}/ci/nova-operator-base/playbooks/install_meta_operators.yaml

or 

just add install_meta_operators.yaml as a pre-run playbook to the nova-base job

**Review by SeanMooney — CHANGES_REQUESTED** [review]

File: ci/nova-operator-base/playbooks/install_meta_operators.yaml (hunk 1 above)
Comment (raukadah) [reply ci/nova-operator-base/playbooks/install_meta_operators.yaml:12]: Done

File: ci/nova-operator-base/playbooks/pre.yaml (hunk 3 above)
Comment (raukadah) [reply ci/nova-operator-base/playbooks/pre.yaml:17]: Thank you @SeanMooney for the review, I will address in the upcoming patch updates.

**Comment by raukadah** [comment]
@SeanMooney Thank you for the valuable comment. I am moving this to draft till it ready for review.

**Comment by raukadah** [comment]
recheck

File: .zuul.yaml [hunk 4]
```diff
@@ -34,9 +34,10 @@
     roles:
       - zuul: github.com/openstack-k8s-operators/ci-framework
     pre-run:
+      - ci/nova-operator-base/playbooks/install_meta_operators.yaml
       - ci/nova-operator-base/playbooks/pre.yaml
-    post-run:
-      - ci/nova-operator-base/playbooks/collect-logs.yaml
+    required-projects:
+      - github.com/openstack-k8s-operators/ci-framework
```
Comment (SeanMooney) [inline .zuul.yaml:40]: ah so this was previosuly a required project in the parent job but now its not ya that makes sesnes.

Comment (SeanMooney) [reply .zuul.yaml:40]: you might want to adrees the zuul job issue by 
adding cifmw_basedir to the vars section below

Comment (raukadah) [reply .zuul.yaml:40]: Done

File: ci/nova-operator-base/playbooks/install_meta_operators.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+---
+- name: Generate install_yamls_make role
+  ansible.builtin.import_playbook: "{{ ansible_user_dir }}/src/github.com/openstack-k8s-operators/ci-framework/ci_framework/playbooks/01-bootstrap.yml"
+  vars:
+    cifmw_install_yamls_out_dir: "~/.ansible/roles"
+    cifmw_installyamls_repos: "{{ ansible_user_dir }}/src/github.com/openstack-k8s-operators/install_yamls"
+
```
[the remark this answers is not in the document]
Comment (raukadah) [reply ci/nova-operator-base/playbooks/install_meta_operators.yaml:8]: Done

**Comment by raukadah** [comment]
Thank you @SeanMooney for all those suggestions. I have added them and now it is failing with [following error](https://review.rdoproject.org/zuul/build/40f28303dcf54aed808dfbd55b75e903/log/job-output.txt#686)
```
ERROR! couldn't resolve module/action 'get_makefiles_env'. This often indicates a misspelling, missing collection, or incorrect module path.
```

It is expected because there is no way to tell zuul to discover[ these action plugins and modules](https://github.com/openstack-k8s-operators/ci-framework/tree/main/ci_framework/plugins/modules).

`[get_makefiles_env](https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/plugins/modules/get_makefiles_env.py)` is used to the env file from install_yamls and [generate_make_tasks](https://github.com/openstack-k8s-operators/ci-framework/blob/main/ci_framework/plugins/modules/generate_make_tasks.py ) generates the install_yamls_makes roles with all make target task files.

In order to fix that in ci-framework, we went with nested ansible-playbook call approach with shell (which is not ansible way).

**Comment by SeanMooney** [comment]
you likely need to follow https://docs.ansible.com/ansible/latest/dev_guide/developing_locally.html#adding-standalone-local-modules-for-all-playbooks-and-roles

**Comment by SeanMooney** [comment]
you might be able to do somethign like this when invoking the role

env:
 ANSIBLE_LIBRARY: "${ANSIBLE_LIBRARY}:~/src/github.com/openstack-k8s-operators/ci-framework/ci_framework/plugins/modules"

**Comment by raukadah** [comment]
recheck

File: ci/nova-operator-base/playbooks/install_meta_operators.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+      vars:
+        cifmw_basedir: "{{ lookup('env', 'HOME') }}/src/github.com/openstack-k8s-operators/ci-framework"
+        cifmw_install_yamls_out_dir: "{{ lookup('env', 'HOME') }}/.ansible/roles"
+        cifmw_installyamls_repos: "{{ lookup('env', 'HOME') }}/src/github.com/openstack-k8s-operators/install_yamls"
+
+- name: Install meta operator
+  hosts: all
+  environment:
+    ANSIBLE_ACTION_PLUGINS: "${ANSIBLE_ACTION_PLUGINS}:~/src/github.com/openstack-k8s-operators/ci-framework/ci_framework/plugins/action:/usr/share/ansible/plugins/action"
+    ANSIBLE_ROLES_PATH: "~/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles"
```
Comment (SeanMooney) [inline ci/nova-operator-base/playbooks/install_meta_operators.yaml:26]: this wont work you removed all the zuul roles from the roles path.

so it wont be able to find install_yamls_makes

**Comment by raukadah** [comment]
recheck

**Comment by raukadah** [comment]
@SeanMooney I have applied all the above suggestion and passed ansible_library and roles_path via environment . But I am still hitting the [same issue](https://logserver.rdoproject.org/00/400/594bf348b412fdc9202189c514aab9faecd9c31f/github-check/nova-operator-compute-kit/0ed9b6e/job-output.txt).
```
TASK [Generate install_yamls_make role name=install_yamls] *********************
2023-06-15 08:47:30.879701 | ERROR! couldn't resolve module/action 'get_makefiles_env'. This often indicates a misspelling, missing collection, or incorrect module path.
2023-06-15 08:47:30.879706 | 
2023-06-15 08:47:30.879710 | The error appears to be in '/var/lib/zuul/builds/0ed9b6ef11d44f5cab26ffdf71894e09/untrusted/project_0/github.com/openstack-k8s-operators/ci-framework/ci_framework/roles/install_yamls/tasks/main.yml': line 27, column 3, but may
2023-06-15 08:47:30.879718 | be elsewhere in the file depending on the exact syntax problem.
2023-06-15 08:47:30.879723 | 
2023-06-15 08:47:30.879729 | The offending line appears to be:
2023-06-15 08:47:30.879734 | 
```
Zuul is not able to discover the ansible modules.Please have a look.
I think we should go with calling ansible-playbook via shell (It is not a ansible thing, but it works).

**Comment by SeanMooney** [comment]
ya i think for now we can go back to calling it via the shell.

this code might get reworked/removed in the future and you have given a decent effort of getting this to work without the sub shell so lets move this forward with the working solution and we can revisit this again later.

**Comment by SeanMooney** [comment]
this si no longer required so closing to clean up the stale prs

**Pull request closed without merging**